scio-tensorflow/src/test/scala/com/spotify/scio/tensorflow/TensorflowSpec.scala

```diff
@@ -18,33 +18,90 @@
 import com.spotify.scio.ContextAndArgs
-import com.spotify.scio.testing.{DistCacheIO, PipelineSpec, TextIO}
+import com.spotify.scio.testing.{PipelineSpec, TextIO}
+import com.spotify.zoltar.tf.TensorFlowModel
 import org.tensorflow._
+import org.tensorflow.example.Example
+
+import scala.io.Source
+
+private object TFGraphJob {
 
-private object TFJob {
   def main(argv: Array[String]): Unit = {
     val (sc, args) = ContextAndArgs(argv)
     sc.parallelize(1L to 10)
-      .predict(args("graphURI"), Seq("multiply"))
-      {e => Map("input" -> Tensors.create(e))}
-      {(r, o) => (r, o.map{case (_, t) => t.longValue()}.head)}
+      .predict(args("graphURI"), Seq("multiply")) { e =>
+        Map("input" -> Tensors.create(e))
+      } { (r, o) =>
+        (r, o.map { case (_, t) => t.longValue() }.head)
+      }
       .saveAsTextFile(args("output"))
-    sc.close()
+    sc.close().waitUntilDone()
   }
 }
 
-private object TFJob2Inputs {
+private object TFGraphJob2Inputs {
+
   def main(argv: Array[String]): Unit = {
     val (sc, args) = ContextAndArgs(argv)
     sc.parallelize(1L to 10)
-      .predict(args("graphURI"), Seq("multiply"))
-      {e => Map("input" -> Tensors.create(e),
-                "input2" -> Tensors.create(3L))}
-      {(r, o) => (r, o.map{case (_, t) => t.longValue()}.head)}
+      .predict(args("graphURI"), Seq("multiply")) { e =>
+        Map("input" -> Tensors.create(e), "input2" -> Tensors.create(3L))
+      } { (r, o) =>
+        (r, o.map { case (_, t) => t.longValue() }.head)
+      }
+      .saveAsTextFile(args("output"))
+    sc.close().waitUntilDone()
+  }
+}
+
+private object TFSavedJob {
+
+  case class Iris(sepalLength: Option[Double],
+                  sepalWidth: Option[Double],
+                  petalLength: Option[Double],
+                  petalWidth: Option[Double],
+                  className: Option[String])
+
+  val Spec: FeatureSpec[Iris] = FeatureSpec
+    .of[Iris]
+    .optional(_.petalLength)(StandardScaler("petal_length", withMean = true))
+    .optional(_.petalWidth)(StandardScaler("petal_width", withMean = true))
+    .optional(_.sepalLength)(StandardScaler("sepal_length", withMean = true))
+    .optional(_.sepalWidth)(StandardScaler("sepal_width", withMean = true))
+
+  def main(argv: Array[String]): Unit = {
+    val (sc, args) = ContextAndArgs(argv)
+    val options = TensorFlowModel.Options.builder.tags(Collections.singletonList("serve")).build
+    val settings = sc.parallelize(List(Source.fromURL(args("settings")).getLines.mkString))
+
+    val collection =
+      sc.parallelize(List(Iris(Some(5.1), Some(3.5), Some(1.4), Some(0.2), Some("Iris-setosa"))))
+
+    Spec
+      .extractWithSettings(collection, settings)
+      .featureValues[Example]
+      .predict(args("savedModelUri"), Seq("linear/head/predictions/class_ids"), options) { e =>
+        Map("input_example_tensor" -> Tensors.create(Array(e.toByteArray)))
+      } { (r, o) =>
+        (r, o.map {
```

Comment:
We probably could use `JTensor` here, but now looking at `JTensor` it doesn't seem like it's `Serializable` :/
https://github.com/spotify/zoltar/issues/104

Reply:
@ravwojdyla yeah that would be nicer i guess!